### PR TITLE
Allow env variables in graphroot and runroot

### PR DESCRIPTION
**Pull request by ashley-cui**
Expand env variables in paths

Part of https://github.com/containers/buildah/issues/2311

Signed-off-by: Ashley Cui <acui@redhat.com>

### CODE REVIEW

**Comment by ashley-cui** [comment]
Might need help with tests, since the test file looks completely foreign to me

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @nalind @mrinaldhillon @saschagrunert PTAL

File: utils.go [hunk 1]
```diff
@@ -271,30 +271,49 @@ func DefaultStoreOptions(rootless bool, rootlessUID int) (StoreOptions, error) {
...
 		}
 	}
 	return storageOpts, nil
 }
+func expandEnvPath(path string, rootlessUID int) (string, error) {
+	if err := validEnvPathFormat(path); err != nil {
+		return path, err
+	}
+	path = strings.Replace(path, "$HOME", homedir.Get(), -1)
+	path = strings.Replace(path, "$UID", strconv.Itoa(rootlessUID), -1)
```
Comment (saschagrunert) [inline utils.go:306]: What about using [os.ExpandEnv](https://golang.org/pkg/os/#ExpandEnv) here?

File: docs/containers-storage.conf.5.md [hunk 2]
```diff
@@ -34,6 +34,11 @@ The `storage` table supports the following options:
   container storage graph dir (default: "/var/lib/containers/storage")
   Default directory to store all writable content created by container storage programs.
 
+  The rootless graphroot path supports three substations:
```
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md:37]: This looks fine here, but if you do "View File" from the '...' in the review page, it's all jammed up into the prior line.

Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md:37]: Maybe it's me pre-tea, but I'm not sure what a "substation" is in this context.  Can you define or provide an example?

Comment (nalind) [reply docs/containers-storage.conf.5.md:37]: I'm guessing `substitutions`.

Comment (ashley-cui) [reply docs/containers-storage.conf.5.md:37]: LOL, I guess I shouldn't copy-paste without reading properly. Will fix

**Pull request merged**